### PR TITLE
Chore/pg update

### PR DESCRIPTION
**Pull request by ubergesundheit**
This PR builds upon https://github.com/Kong/charts/pull/541 with changes:

- Rebased onto current `main` branch
- Change the `apiVersion` in `Chart.yaml` to `v2` as suggested by @edmund-wagner
- Update to postgresql chart 11.1.15. It contains a fix in regards to RBAC resources
- Update CI to use newer `ct` (3.5.1), `helm` (3.8.1) and python `3.9.7` versions
- Update ct files to reflect changed values in postgresql chart
- Add default value to use postgresql 13.6

#### What this PR does / why we need it:

The old postgresql subchart does not support RBAC.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] PR is based off the current tip of the `main` branch.
- [x] Changes are documented under the "Unreleased" header in CHANGELOG.md
- [x] Commits follow the [Kong commit message guidelines](https://github.com/Kong/kong/blob/master/CONTRIBUTING.md#commit-message-format)

### CODE REVIEW

File: charts/kong/templates/_helpers.tpl [hunk 1]
```diff
@@ -778,10 +778,10 @@ the template that it itself is using form the above sections.
 
 {{- if .Values.postgresql.enabled }}
   {{- $_ := set $autoEnv "KONG_PG_HOST" (include "kong.postgresql.fullname" .) -}}
-  {{- $_ := set $autoEnv "KONG_PG_PORT" .Values.postgresql.service.port -}}
+  {{- $_ := set $autoEnv "KONG_PG_PORT" .Values.postgresql.service.ports.postgresql -}}
   {{- $pgPassword := include "secretkeyref" (dict "name" (include "kong.postgresql.fullname" .) "key" "postgresql-password") -}}
   {{- $_ := set $autoEnv "KONG_PG_PASSWORD" $pgPassword -}}
-{{- else if .Values.postgresql.enabled }}
+{{- else if eq .Values.env.database "postgres" }}
```
Comment (ubergesundheit) [inline charts/kong/templates/_helpers.tpl:785]: Restoring https://github.com/Kong/charts/commit/a3ed08a2f779b37cc7c1aa3f61c4dacf4ce98d6a

**Comment by ubergesundheit** [comment]
I'll change to a Draft PR for now, sorry for the noise

**Comment by ubergesundheit** [comment]
Unsure what happened here, but latest error looks unrelated

@rainest @shaneutt

```
Error: module 'openssl.rand' not found:No LuaRocks module found for openssl.rand
	no field package.preload['openssl.rand']
	no file '/opt/openssl/rand.lua'
	no file '/opt/openssl/rand/init.lua'
	no file './openssl/rand.lua'
	no file './openssl/rand/init.lua'
	no file '/usr/local/openresty/site/lualib/openssl/rand.ljbc'
	no file '/usr/local/openresty/site/lualib/openssl/rand/init.ljbc'
	no file '/usr/local/openresty/lualib/openssl/rand.ljbc'
	no file '/usr/local/openresty/lualib/openssl/rand/init.ljbc'
	no file '/usr/local/openresty/site/lualib/openssl/rand.lua'
	no file '/usr/local/openresty/site/lualib/openssl/rand/init.lua'
	no file '/usr/local/openresty/lualib/openssl/rand.lua'
	no file '/usr/local/openresty/lualib/openssl/rand/init.lua'
	no file './openssl/rand.lua'
	no file '/usr/local/openresty/luajit/share/luajit-2.1.0-beta3/openssl/rand.lua'
	no file '/usr/local/share/lua/5.1/openssl/rand.lua'
	no file '/usr/local/share/lua/5.1/openssl/rand/init.lua'
	no file '/usr/local/openresty/luajit/share/lua/5.1/openssl/rand.lua'
	no file '/usr/local/openresty/luajit/share/lua/5.1/openssl/rand/init.lua'
	no file '/home/kong/.luarocks/share/lua/5.1/openssl/rand.lua'
	no file '/home/kong/.luarocks/share/lua/5.1/openssl/rand/init.lua'
	no file '/usr/local/openresty/site/lualib/openssl/rand.so'
	no file '/usr/local/openresty/lualib/openssl/rand.so'
	no file './openssl/rand.so'
	no file '/usr/local/lib/lua/5.1/openssl/rand.so'
	no file '/usr/local/openresty/luajit/lib/lua/5.1/openssl/rand.so'
	no file '/usr/local/lib/lua/5.1/loadall.so'
	no file '/home/kong/.luarocks/lib/lua/5.1/openssl/rand.so'
	no file '/usr/local/openresty/site/lualib/openssl.so'
	no file '/usr/local/openresty/lualib/openssl.so'
	no file './openssl.so'
	no file '/usr/local/lib/lua/5.1/openssl.so'
	no file '/usr/local/openresty/luajit/lib/lua/5.1/openssl.so'
	no file '/usr/local/lib/lua/5.1/loadall.so'
	no file '/home/kong/.luarocks/lib/lua/5.1/openssl.so'
```

**Comment by rainest** [comment]
The OpenSSL error is fixed by https://github.com/Kong/kong/pull/8429, but I'm unsure of the release timeline for it; checking to see. If it's soon-ish, we should just wait for that to be out. If not, we'd need to pin the Postgres version to 13 for now.

Otherwise LGTM.

**Comment by ubergesundheit** [comment]
Ah yes, have seen that before, now I remember. I was using enterprise kong-gateway container image where postgres 14 seems to work. Thanks for the suggestion :)

I think this is now ready for review

**Comment by rainest** [comment]
One last bit (also waiting on a response from the Gateway devs to see if we'd just be able to go to 14 in the near future): could you remove the tarball (both the original 8.x one and the new 11.x one) from `charts/kong/charts` entirely? There shouldn't actually be any reason for it to be there; it got checked in accidentally from someone's working copy that'd run `helm dependency update` while working directly from the repo.

**Comment by ubergesundheit** [comment]
> could you remove the tarball (both the original 8.x one and the new 11.x one) from charts/kong/charts entirely?

done in [0cbf680](https://github.com/Kong/charts/pull/572/commits/0cbf680537c9b7ade03ece4d2e560acc891936b9)

**Comment by ubergesundheit** [comment]
Mh looks like the archive is required ( see failed tests)

**Comment by ubergesundheit** [comment]
maybe https://github.com/Kong/charts/pull/572/commits/ec0a4780d603653a1b96169e9aa56d5ea45860be fixes the tests in CI

**Comment by rainest** [comment]
Gateway team have informed me that the 14 fix should be out imminently, so will wait a few days and then strip out the limit here barring some delay.

**Comment by ubergesundheit** [comment]
Thanks! Can you give an estimate on how long we have to wait?

**Comment by rainest** [comment]
Should be within a week, or rather, if it's not, will merge this as-is.

**Comment by ubergesundheit** [comment]
Thanks, looking forward :+1:

**Comment by rainest** [comment]
Actually, whatever, we need to make other unrelated changes to bump the controller and chart version for the next release anyway, so will go ahead and merge this as-is and remove the image default then assuming the Kong release is ready.

**Review by rainest — APPROVED** [review]

**Comment by ubergesundheit** [comment]
Thank you :)

**Comment by rainest** [comment]
Thank you for your contribution. Please fill out the following form to claim your contributor swag:
https://github.com/Kong/kong/blob/master/CONTRIBUTING.md#contributor-t-shirt.

**Comment by ubergesundheit** [comment]
Nice, thank you :) Will do

**Pull request merged**